### DISCUSSION TITLE
Did mobile version support memo?

### DISCUSSION THREAD

**Discussion by katopz** [phantom/docs#97] [General]
Refer to https://github.com/solana-labs/solana-pay/blob/master/SPEC.md#memo
But after I try it seem like `memo` is not support and there's no information about this on docs.

<img width="1020" alt="image" src="https://github.com/phantom/docs/assets/97060/09362fa2-9b68-4ae3-b877-dfc0b3cec088">

It would be nice if we support `memo` so we can know what this tx is reference to.

IMHO: This is another show stopper for simple integration use case (just pay without sign-in) [because redirect also not support](https://github.com/orgs/phantom/discussions/84) 

which mean the only way to do this is include plenty of lib for build the tx and force user to connect wallet and pay (2 steps) or use in-app webview (still needed to sign-in implementation) while we can use reference_id or redirect_url callback for payment (1 step).

Thanks

**Comment by adamdelphantom** [comment]
Hey @katopz,

I understand this is not included in the docs. Have you attempted passing a transaction with memo field and seeing if it is accessible? I don't have a way to readily test right now but I believe we may already support the memo field.

**Reply by katopz** [reply]
Oh it's not transaction case, I tend to use just 1 line deeplink so no setup or `web3.js` needed here.

### refer to solana example

```
solana:mvines9iiHiQTysrwkJjGf2gb9Ex9jXJX8ns3qwf2kN?amount=1&label=Michael&message=Thanks%20for%20all%20the%20fish&memo=OrderId12345
```

- Yes, I attempt to pass the `memo` param many times.
- No, I can't make it work.
- No `memo` on confirmation screen.
- No `memo` related in final tx.
- The final tx is just transfer both sol and spl.

### And that 1 line link is the point.
- Yes, I can build my own tx with memo then ask wallet to `signAndSendTransaction` which I do that for living via js, Rust, wasm.
- But no, I'm here asking for `memo` support so I (and other thousand of developer) don't needed to use any web3 library at all.(say mass adoption = easy integration = no `node_modules` = just 1 line and phantom will take care of it = we love phantom)

I can understand that transfer with memo can be complicate and somehow not wallet side concern.
But to become the wallet that we can 100% love and trust, we should support this simple real world use case so newcomer can just use it with 1 line I think.

And this doable, even has a spec for it so please support it or maybe just say something about it in docs if it's not or plan to.

Thanks for your time.

**Comment by samcode206** [comment]
Confirming this is still broken as of September 2026, with on-chain evidence. Given a Solana Pay transfer request of the form `solana:<recipient>?amount=0.009528&memo=<60-char base58 string>`:

**Phantom mobile** (deep link opened on iOS device):
https://solscan.io/tx/XuwiynfSY81gNvtmdcMDJvWL2kSQiQg5BW5o6guR3JYzwHSsUVPYJR6w8caiDMXEXRAMmSYLCneVW2gQnrKfYrT
The transaction contains only ComputeBudget instructions and a SystemProgram transfer — **no Memo instruction**. The `memo` param is silently discarded.

**Solflare mobile**, same URI format, same sender wallet, same recipient:
https://solscan.io/tx/59gDHkXw8NcMNiDgwD63BBGDAnzmaQrDcphQkaBHJykuCaQuaGM6HC742MPbzfya3A8uA7zYJ2iD8LzMduXmEWVQ
Memo instruction included immediately before the transfer, exactly per spec.

The [Solana Pay spec](https://docs.solanapay.com/spec#memo) requires that when the `memo` field is provided, the wallet **must** include it in an SPL Memo instruction in the payment transaction. We use the memo to associate payments with invoices; because Phantom drops it, payments arrive at our acceptance address with no way to link them to the paying user's invoice. the payment silently fails to apply and requires manual recovery, through no fault of the user.
